### DISCUSSION TITLE
Null-propagating invocation

### DISCUSSION THREAD

**Discussion by dylanrafael05** [dotnet/csharplang#4998] [Language Ideas] [answered]
Simply to reduce clutter.
# Concept and reasoning
The execution of a possibly-null delegate or event is a common task to perform in any program which uses either of the two concepts. In order to make this task slightly less wordy, and prevent the user from being required to know of the existence of `Delegate.Invoke`, null-propagating invocation could be introduced:
```cs
public class MyEventInvoker 
{
     public event Action myEvent;
     public void InvokeEvent() 
     {
          myEvent?();
     }
}
```
Similar to null-propagating member access, a delegate with a return type of `T` would return `T?` if `T` is a value-type when called using `?()`.
### Behaviour of implementation
`?()` will behave identically to `?.Invoke()` in that no delegate or event code is executed if said item evaluates to `null`, and any returned value will be represented as `null`.
### Drawbacks
Language complexity _blahblahblah_.
Possible confusion with ?[].
### Alternatives
`?.Invoke()`

**Comment by khm1600** [comment, marked as answer]
https://github.com/dotnet/csharplang/issues/3257
